code/game/machinery/suit_storage_unit.dm

```diff
@@ -539,19 +524,15 @@
 
 ////////
 
-/obj/machinery/suit_storage_unit/attack_hand(mob/user as mob)
+/obj/machinery/suit_storage_unit/attack_hand(mob/user)
 	var/dat
-	if(shocked)
-		if(shock(usr, 100))
+	if(shocked && !(stat & NOPOWER))
+		if(shock(user, 100))
 			return
-	if(..())
-		return
-	if(stat & NOPOWER)
-		return
```

Comment:
1) Without the parent call here, you skip a bunch of stuff, like an incapacitated check, a brain damage check, adding fingerprints of the mob to the unit, etc.
2) Without some sort of power check like this, you can open the UI of the unit while it doesn't have power, which I think is a bit weird.